CHANGELOG.md

```diff
@@ -34,6 +34,36 @@ New features:
   Now, these types' kind signatures are displayed above their declarations
   in their docs, similar to what one would see in the source code.
 
+  But that's not all!
+
+  The two below types do not have explicit kind signatures, but the compiler
+  infers what their kind signature is anyway. "Interesting" kind signatures
+  will be displayed but "uninteresting" kind signatures (even if they are
+  explicit) will be removed.
+
+  ```purescript
+                                                  {-
+  data Foo :: (Type -> Type) -> Type -> Type      -}
+  data Foo f a = Foo (f a)
+                                                  {-
+  data Tuple :: Type -> Type -> Type              -}
+  data Tuple a b = Tuple a b
+  ```
+
+  The docs will include the kind signature for `Foo`, but will
+  intentionally remove the kind signature for `Bar`, even if `Bar`'s
+  kind signature was explicitly defined by the developer. `Bar`'s
+  kind signature is considered "uninteresting" because all of the
+  type parameters have kind `Type`. `Foo`, on the other hand, has
+  at least one type parameter, `f`, whose kind is something other than
+  `Type`. Thus, `Foo`'s inferred kind signature will be displayed
+  in its docs.
+
+  An "uninteresting" kind signature is one that follows this form:
```

Comment:
I think we should say "kind" rather than "kind signature" here. A kind is something like `Type`, `Type -> Type`, or `forall k. k -> Type`. A kind signature is more than that: it's a declaration which tells you that a particular data type or type synonym or class has a particular kind, such as `data Tuple :: Type -> Type -> Type`.

Reply:
Mm... Makes sense. I've rewritten parts of the changelog entry to keep this in mind.